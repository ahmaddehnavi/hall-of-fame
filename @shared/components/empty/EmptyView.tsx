import {BaseResource, Row} from '@shared';
import * as React from 'react';
import {Text} from 'react-native';

export class EmptyView extends React.Component<{ style?, message: string }> {
    render() {
        return (
            <Row style={{
                backgroundColor: 'rgba(255,255,255,.5)',
                alignItems: 'center',
                justifyContent: 'center',
                flex: 1,
            }}>
                <Text>
                    {
                        this.props.message || 'No Items!'
                    }
                </Text>
            </Row>
        )
    }
}