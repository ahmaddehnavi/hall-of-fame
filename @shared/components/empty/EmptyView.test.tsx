import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {EmptyView} from './EmptyView';

describe('EmptyView Tests', () => {

    it('should match snapshot', () => {
        let tree = renderer.create(
            <EmptyView
                message={'no item'}
                style={{padding: 16}}
            />
        );
        expect(tree).toMatchSnapshot();
    });

});
