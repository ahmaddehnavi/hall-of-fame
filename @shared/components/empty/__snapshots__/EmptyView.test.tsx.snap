// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyView Tests should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255,255,255,.5)",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text>
    no item
  </Text>
</View>
`;
